File: src/dataset/typescript/pangram_exercism-wackerow/claude/ClaudeSonnet4_pangram_v2.ts
export default class {
  constructor(private sentence: string) {}

  isPangram(): boolean {
    const seen = new Set<string>();
    
    for (let i = 0; i < this.sentence.length; i++) {
      const char = this.sentence[i].toLowerCase();
      if (char >= 'a' && char <= 'z') {
        seen.add(char);
        if (seen.size === 26) {
          return true;
        }
      }
    }
    
    return false;
  }
}

File: src/dataset/c/pangram_exercism-yorange1/claude/ClaudeSonnet4_pangram_v2.c
#include "pangram.h"

#include <ctype.h>
#include <stddef.h>
#include <stdint.h>

#define FLAGS_EXCEPT (0x03FFFFFF)

bool is_pangram(const char *sentence)
{
    if (NULL == sentence)
        return false;

    uint32_t flags = 0;
    const char *ptr = sentence;

    while (*ptr && flags != FLAGS_EXCEPT)
    {
        if (isalpha(*ptr))
        {
            flags |= 1 << (tolower(*ptr) - 'a');
        }
        ptr++;
    }

    return (flags == FLAGS_EXCEPT);
}